File: client/src/apis/phobert.ts
import { PHOBERT_CONFIG } from ".";

export const usingPHOBERT = {
    usingCNN: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/cnn', {title: title, content: content})
    },
    usingLSTM: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/lstm', {title: title, content: content})
    },
    usingBILSTM: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/bilstm', {title: title, content: content})
    },
    usingENSEMBLE_CNN_BILSTM: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/ensemble_cnn_bilstm', {title: title, content: content})
    },
    usingFUSION_CNN_BILSTM: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/fusion_cnn_bilstm', {title: title, content: content})
    },
    usingENSEMBLE_CNN_BIGRU: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/ensemble_cnn_bigru', {title: title, content: content})
    },
    usingFUSION_CNN_BIGRU: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/fusion_cnn_bigru', {title: title, content: content})
    },
    usingGRU: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/gru', {title: title, content: content})
    },
    usingBIGRU: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/bigru', {title: title, content: content})
    },
    usingTRANSFORMER: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/transformer', {title: title, content: content})
    },
    usingLOGISTIC: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/logistic', {title: title, content: content})
    },
    usingSGD: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/sgd', {title: title, content: content})
    },
    usingRANDOM_FOREST: (title: string, content: string) => {
        return PHOBERT_CONFIG.post('/random_forest', {title: title, content: content})
    }
}

File: server/phobert.py
from flask import Blueprint, request, jsonify
from appService import getRatingFromModel
from constants import *

phobert = Blueprint('phobert', __name__, url_prefix="/api/v1/phobert")

@phobert.post('/cnn')
def using_cnn():
    req = request.get_json()
    title = req.get('title')
    content = req.get('content')

    res = getRatingFromModel(title, content, CNN_MODEL, PHOBERT_METHOD).tolist()

    return jsonify({'prediction': res})

@phobert.post('/lstm')
def using_lstm():
    req = request.get_json()
    title = req.get('title')
    content = req.get('content')

    res = getRatingFromModel(title, content, LSTM_MODEL, PHOBERT_METHOD).tolist()

    return jsonify({'prediction': res})

@phobert.post('/bilstm')
def using_bilstm():
    req = request.get_json()
    title = req.get('title')
    content = req.get('content')

    res = getRatingFromModel(title, content, BILSTM_MODEL, PHOBERT_METHOD).tolist()

    return jsonify({'prediction': res})

@phobert.post('/ensemble_cnn_bilstm')
def using_ensemble_cnn_bilstm():
    req = request.get_json()
    title = req.get('title')
    content = req.get('content')

    res = getRatingFromModel(title, content, ENSEMBLE_CNN_BILSTM, PHOBERT_METHOD).tolist()

    return jsonify({'prediction': res})

# @phobert.post('/fusion')
# def using_fusion_cnn_bilstm():
#     req = request.get_json()
#     title = req.get('title')
#     content = req.get('content')

#     res = getRatingFromModel(title, content, FUSION_CNN_BILSTM).tolist()

#     return jsonify(res)
